app/models/user.rb

```diff
@@ -81,6 +84,25 @@ class User < ActiveRecord::Base
     end
   }
 
+  def require_2sv?
+    return false if otp_secret_key.present?
+
+    if deferred_2sv_at?
+      deferred_2sv_at < 24.hours.ago
+    else
+      super
+    end
+  end
+
+  def defer_two_step_verification
+    transaction do
+      self.require_2sv = false
```

Comment:
Agreed. It also says "and the user can't turn it off", should we add that functionality later.